subsys/bluetooth/host/hci_core.c

```diff
@@ -3657,6 +3668,37 @@ const char *bt_get_name(void)
 #endif
 }
 
+uint16_t bt_get_appearance(void)
+{
+#if defined(CONFIG_BT_DEVICE_APPEARANCE_DYNAMIC)
+	return bt_dev.appearance;
+#else
+	return CONFIG_BT_DEVICE_APPEARANCE;
+#endif
+}
+
+int bt_set_appearance(uint16_t new_appearance)
+{
+#if defined(CONFIG_BT_DEVICE_APPEARANCE_DYNAMIC)
+	if (IS_ENABLED(CONFIG_BT_SETTINGS)) {
+		int err = settings_save_one("bt/appearance",
+			&new_appearance, sizeof(new_appearance));
+
+		if (err) {
+			BT_ERR("Unable to persistently store setting bt/appearance "
+			       "(err %d). Ensure bt_enable and settings_load is called "
+				   "before this function.", err);
+		}
```

Comment:
We don't want to return this error in the function?

Reply:
Not sure. `set_name` does not abort when storing fails. I copied the behavior for consistency. If there were no precedent, I would have returned the error here.

Reply:
I guess it makes sense to just log this, as name is technically set even if fails to store it in storage.